enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,328 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes converging two somewhat different bare-metal
+provisioning flows in OpenShift:
+
+- *Standard* Metal3 flow used in bare-metal IPI and [baremetal-operator][bmo]
+  (BMO). Uses the [Ironic agent][ipa] (also known as ironic-python-agent or
+  simply IPA) for writing the CoreOS image.
+
+- ZTP (Zero-Touch Provisioning) flow using the so-called *live ISO* Ironic and
+  baremetal-operator feature, the assisted service and the assisted agent.
+
+## Motivation
+
+Having two different provisioning flows comes with significant maintenance and
+development costs:
+
+- Some features, namely firmware and RAID configuration, require using the
+  Ironic agent, which is not present in the *live ISO* flow.
+
+- Supporting (i)PXE boot with the *live ISO* flow requires creating new API
+  in baremetal-operator specifically for this purpose.
+
+- In the *live ISO* flow, it is not possible to notify Ironic (and thus
+  baremetal-operator) when the actual installation finishes. From the
+  perspective of Metal3 components, the installation is finished once the ISO
+  is connected. As a result, Ironic cannot disconnect the ISO from the
+  machine's BMC, nor can it update the BMC's boot device settings properly.
+
+- The *live ISO* flow disables Ironic *inspection* process and has to simulate
+  it by updating the `BareMetalHost` based on the assisted service's inspection
+  process. This creates an unnessesary coupling between the assisted service
+  and the baremetal-operator's hardware inventory format (which, in turn, is
+  derived from the Ironic's inventory format).
+
+- The *standard* flow is highly [extensible][hardware managers] via the
+  [ironic-agent-image][ironic-agent] container image. This extensibility does
+  not apply to the *live ISO* flow since it does not use the Ironic agent.
+
+- Any different path through the `BareMetalHost` state machine makes it
+  exponentially more difficult for upstream developers to reason about the
+  effects of any changes. In addition, the *live iso* flow is not tested
+  upstream by the Metal3 project.
+
+### Goals
+
+- ZTP follows the *standard* Metal3 flow instead of *live ISO*.
+- (i)PXE boot, firwmare and RAID settings work for ZTP without introducing
+  special API in Metal3 components.
+- Continue to allow non-ZTP assisted deployments without depending on Metal3.
+- Not depend on assisted components in the core metal platform
+
+### Non-Goals
+
+- Converging two agent implementations (Ironic and assisted) into a single one.
+- Supporting per-host network configuration through the BMH
+  PreprovisioningNetworkData API in the ZTP case.
+- Performing firmware config changes only after the assisted service runs host
+  validations.
+
+## Proposal
+
+- Building CoreOS images for the *standard* flow is handled by
```

Comment:
This first sentence needs some work, because it's directly contradicted by the second.

Reply:
Could you elaborate? Here I'm contrasting the standard flow with the ZTP flow.